### PR TITLE
refactor occupations, add green topic

### LINKED ISSUE #38: Finalising/adding to occupation greenness datasets

### PR DESCRIPTION
**Pull request by lizgzil**
---

Closes #38 #23 

# Description

- Adds ONET green topic to the occupation measures
- Refactors occupations (inc moving a lot of the code out of [green_measures.py](https://github.com/nestauk/dap_prinz_green_jobs/pull/52/files#diff-198d3c490c352b47b17547003a8acc4d1147c6a16abcf19268537aadc082f1b6) and into [occupations_measures_utils.py](https://github.com/nestauk/dap_prinz_green_jobs/pull/52/files#diff-6c04a650eccc6cac70ea757816804dfc272dc0be4b5787869dfd88daec3e2b23))

# Instructions for Reviewer

Run
```
pytest
```

Also try out the new occupation class functionality by running:
```
from dap_prinz_green_jobs.pipeline.green_measures.occupations.occupations_measures_utils import OccupationMeasures

om = OccupationMeasures()
om.load()

unique_job_titles = ["Data Scientist", "Nurse", "Key Stage 4 teacher", "Pharmacist", "Biologist"]
job_title_2_match = om.precalculate_soc_mapper(unique_job_titles)
om.get_green_measure_for_job_title("Data Scientist")
```

This should give you the latest green occupation measures + SOC information, 

```
{'GREEN CATEGORY': 'Non-Green', 'GREEN/NOT GREEN': 'Non-green', 'GREEN TIMESHARE': 12.9, 'GREEN TOPICS': 21, 'SOC': {'SOC_2020_EXT': '2433/02', 'SOC_2020': '2433', 'SOC_2010': '2114'}}
```
# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

File: dap_prinz_green_jobs/tests/test_occupation_measures.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import pytest
```
Comment (india-kerle) [inline dap_prinz_green_jobs/tests/test_occupation_measures.py:1]: omg haven't reviewed but INSPIRED by this test script

Comment (lizgzil) [reply dap_prinz_green_jobs/tests/test_occupation_measures.py:1]: 💅

Comment (india-kerle) [reply dap_prinz_green_jobs/tests/test_occupation_measures.py:1]: this ran for me perfectly btw!

File: dap_prinz_green_jobs/pipeline/green_measures/green_measures.py [hunk 2]
```diff
@@ -185,45 +166,11 @@ def get_occupation_measures(self, job_advert: Dict[str, str]) -> List[dict]:
...
-            else:
-                occ_green_measures_list.append(
-                    {
-                        "GREEN CATEGORY": None,
-                        "GREEN/NOT GREEN": None,
-                        "GREEN TIMESHARE": None,
-                        "SOC": None,
-                    }
-                )
+        job_title_2_match = self.om.precalculate_soc_mapper(unique_job_titles)
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/green_measures.py:170]: nice, when I ran this on the example you suggested, it ran perfectly.

File: dap_prinz_green_jobs/pipeline/green_measures/occupations/occupations_measures_utils.py [hunk 3]
```diff
@@ -30,18 +34,263 @@ def clean_job_title(job_title: str) -> str:
...
+        lambda x: gla_mapper.get(x)[0] if gla_mapper.get(x) else None
+    )
+    green_topics["SOC_2010_name_GLA"] = green_topics["Code"].apply(
+        lambda x: gla_mapper.get(x)[1] if gla_mapper.get(x) else None
+    )
+
+    # Use GLA if we don't find it
+    # Of the 1354 ONET occupations, we find 2010 SOC for 1081 of them (80%)
+    # Of the 1354 ONET occupations, GLA find 2010 SOC for 721 of them (53%)
+    # We could recover 141 of our 273 not found ones by using GLA's
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/occupations/occupations_measures_utils.py:85]: nice

**Review by india-kerle — APPROVED** [review]
looks good! I was able to run the tests and was able to use the `OccupationMeasures` class seamlessly.

**Comment by lizgzil** [comment]
thanks @india-kerle !

**Pull request merged**